### PR TITLE
kernel: Enable SGX in experimental kernel.

### PR DESCRIPTION
**Pull request by jcvenegas**
Enable Intel SGX support in experimental kernel.

Upgrade Linux kernel to latest stable release.

Allow build experimental kernel from kata-deploy.

```
 make kernel-experimental-tarball
```

Fixes: #2518

@mythi  @eadamsintel

### CODE REVIEW

File: tools/packaging/kernel/build-kernel.sh [hunk 1]
```diff
@@ -443,7 +454,7 @@ main() {
 				set -x
 				;;
 			e)
-				experimental_kernel="true"
+				build_type="experimental"
```
Comment (cmaf) [inline tools/packaging/kernel/build-kernel.sh:457]: @jcvenegas do we need to keep the `-e` flag if we can now specify experimental with `-b`?

**Review by cmaf — COMMENTED** [review]
Thanks @jcvenegas, I have question about build types below

File: versions.yaml [hunk 2]
```diff
@@ -153,14 +153,12 @@ assets:
...
     url: "https://cdn.kernel.org/pub/linux/kernel/v5.x/"
-    uscan-url: >-
-      https://mirrors.edge.kernel.org/pub/linux/kernel/v5.x/linux-(5\.4\..+)\.tar\.gz
     version: "v5.10.25"
 
   kernel-experimental:
     description: "Linux kernel with virtio-fs support"
     url: "https://cdn.kernel.org/pub/linux/kernel/v5.x/"
-    tag: "v5.10.25"
+    tag: "v5.13.10"
```
Comment (mythi) [inline versions.yaml:161]: I believe you need `patches/v5.13.x/` with this change? At least for me `setup` did not finish properly (it did not generate `.config`)  until I added `patches/v5.13.x/no_patches`

**Review by mythi — CHANGES_REQUESTED** [review]
`v5.13.x` patches (or `no_patches`) are needed.

minor: noticed a typo in the script: "emtpy".

File: tools/packaging/kernel/build-kernel.sh (hunk 1 above)
Comment (jcvenegas) [reply tools/packaging/kernel/build-kernel.sh:457]: @cmaf I was about to remove it, but there are some scripts that still use them, I prefer to keep it as a legacy option.

**Review by mythi — APPROVED** [review]
tested and works now, LGTM

**Comment by jcvenegas** [comment]
Kernel seems to build OK, but experimental qemu job is broken,  https://github.com/kata-containers/kata-containers/pull/2502 will fix it

**Comment by mythi** [comment]
@jcvenegas needs rebasing. looks like #2502 is getting ready too

**Review by mythi — APPROVED** [review]

**Comment by mythi** [comment]
@jcvenegas thanks. do you know how can help merging?

**Review by devimc — APPROVED** [review]

**Review by cmaf — APPROVED** [review]

**Pull request merged**